packages/foundry/contracts/Reputation.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./Events.sol";
+
+contract Reputation is ERC20 {
+  address public immutable spotlightContract;
+  uint256 private constant DECAY_INTERVAL = 15 minutes;
+  uint256 private constant DECAY_RATE = 1; // 1% decay every 15 minutes
+  mapping(address => uint256) private lastDecayTime;
+
+  constructor(address _spotlightContract) ERC20("Reputation", "RPT") {
+    require(_spotlightContract != address(0), "Spotlight address cannot be zero");
+    spotlightContract = _spotlightContract;
+  }
+
+  /**
+   * @dev Internal function to issue tokens
+   * @param receiver The address that will receive the tokens
+   * @param amount The number of tokens to issue (will be multiplied by 10^decimals)
+   */
+  function _issueToken(address receiver, uint256 amount) internal {
+    require(receiver != address(0), "Cannot issue to zero address");
+
+    uint256 tokenAmount = amount * 10 ** decimals();
```

Comment:
A TODO for future us: we should probably leverage `openzepplin` to prevent instances of overflows when doing any math operations :grimacing: 

https://docs.openzeppelin.com/contracts/2.x/api/math

Added to the trello board :heart:

Reply:
It seems that `SafeMath` has been removed from `openzepplin`'s current API, as it is implemented language-level: https://stackoverflow.com/questions/70074736/do-you-need-safemath-in-solidity-version-0-8-and-if-not-can-you-still-import

Reply:
Nice!

Removed from the trello board :sunglasses: